src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java

```diff
@@ -298,4 +319,33 @@ private static ConfigDef createConfigDef() {
     MongoSinkTopicConfig.BASE_CONFIG.configKeys().values().forEach(configDef::define);
     return configDef;
   }
+
+  /**
+   * This method is used in our implementation of the {@link ConfigDef#validateAll(Map)} method
+   * because the original implementation silently disregards any undefined configuration properties.
+   *
+   * @param propertyNames Either normal configuration property names or property names prefixed with
+   *     {@link MongoSinkTopicConfig#TOPIC_OVERRIDE_PREFIX} and a topic name, as specified <a
+   *     href="https://docs.mongodb.com/kafka-connector/master/sink-connector/configuration-properties/topic-override/">here</a>.
+   * @see #OBSOLETE_CONFIGS
+   */
+  static void logObsoleteProperties(final Collection<String> propertyNames) {
+    propertyNames.stream()
+        .filter(MongoSinkConfig::isObsolete)
+        .forEach(
+            obsoletePropertyName ->
+                LOGGER.warn(
+                    "The configuration property {} is obsolete. Remove it as it has no effect.",
```

Comment:
> we should tell the user to set retryWrites=true

I addressed this in the [scope document](https://docs.google.com/document/d/1V0gfwdbfQzseH60yMEbFoHUO-vM3NcZNUknp6C0gKzI/edit):
> The connector requires the Java driver 4.3.x, which has [retries enabled by default](https://jira.mongodb.org/browse/JAVA-3198), meaning that no changes are needed to utilize the mechanism.

I updated the message to reflect that a user does not need to do anything beyond removing the obsolete configuration properties.

Reply:
I still think the message will lead to customer questions and therefore add support burden. It tells users what to do and not the why its no longer required.

Also although retries are enabled by default - if a user previously set `retryWrites=false` thinking the Kafka connector retry mechanism is preferred then they do have to do some action to enable retryable writes.

While the logic for obsolete messages is clean and DRY, the warn message is only needed if `max.num.retries > 1`  and `max.num.retries` defaults to 1.

Reply:
> tells users what to do and not the why its no longer required.
> if a user previously set retryWrites=false thinking the Kafka connector retry mechanism is preferred then they do have to do some action to enable retryable writes.

Updated the warning message to address the above.

> the warn message is only needed if max.num.retries > 1

I think that warning a user that he uses an obsolete configuration option is useful regardless of the option value. Additionally, the proposed logic
- is a complication just for the sake of not showing a warning message in some situations, despite the user specifying values for obsolete configuration properties;
- requires us to read and parse configuration values directly from `Map<String, String>` in `validateAll` as opposed to reading the values via the `AbstractConfig` API.

Accordingly, I find the proposed logic harmful rather than helpful.